### PR TITLE
ArrayBasedStack - Fix bugs in Peek(), Pop() to show stack empty

### PR DESCRIPTION
**Pull request by NageshAndani**
Fixed bugs in ArrayBasedStack implementation (ArrayBasedStack.cs)
- Renamed Count variable as Top to be in line with Stack's terminology
- Added code to check if stack is empty before performing Pop() or Peek()

- [x] I have performed a self-review of my code
- [x] My code follows the style guidelines of this project
- [x] I have added comments to hard-to-understand areas of my code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] I have made corresponding changes to the comments
- [ ] I have made corresponding changes to the README.md

### CODE REVIEW

File: DataStructures/ArrayBasedStack/ArrayBasedStack.cs [hunk 1]
```diff
@@ -70,30 +75,45 @@ public int Capacity
...
         /// <summary>
         /// Returns the item at the top of the <see cref="ArrayBasedStack{T}"/> without removing it.
         /// </summary>
         /// <returns>The item at the top of the <see cref="ArrayBasedStack{T}"/>.</returns>
-        public T Peek() => stack[count - 1];
+        public T Peek()
+        {
+            if (top == -1)
+            {
+                throw new ApplicationException(StackEmptyErrorMessage);
```
Comment (siriak) [inline DataStructures/ArrayBasedStack/ArrayBasedStack.cs:99]: image

Comment (siriak) [reply DataStructures/ArrayBasedStack/ArrayBasedStack.cs:99]: Try switching to InvalidOperationException

File: DataStructures/ArrayBasedStack/ArrayBasedStack.cs [hunk 2]
```diff
@@ -70,30 +75,45 @@ public int Capacity
...
         /// </summary>
         /// <returns>The item removed from the top of the <see cref="ArrayBasedStack{T}"/>.</returns>
         public T Pop()
         {
-            var item = stack[count - 1];
-            count = count - 1;
-            return item;
+            if (top == -1)
+            {
+                throw new ApplicationException(StackEmptyErrorMessage);
```
Comment (siriak) [inline DataStructures/ArrayBasedStack/ArrayBasedStack.cs:113]: And here

**Review by siriak — DISMISSED** [review]
Looks good, only need to change the exception type to follow Codacy's rules :)

File: DataStructures/ArrayBasedStack/ArrayBasedStack.cs (hunk 1 above)
Comment (NageshAndani) [reply DataStructures/ArrayBasedStack/ArrayBasedStack.cs:99]: Thanks for the review. Let me change this and commit.

**Review by siriak — APPROVED** [review]
Looks good, thanks!

**Pull request merged**